Tests/IntegrationTests/Fixtures/1Dimension/issue-3184.e2e.js

```diff
@@ -0,0 +1,150 @@
+        //
+        await t
+            .expect(Selector('[role="alert"][class*="error"]').exists)
+            .notOk();
+    }
+);
+
+test(
+    'Scenario #2: Moved nodes do not just disappear after discarding the move change',
+    async t => {
+        //
+        // Select 🗋 MultiA and 🗋 MultiB, then drag both documents INTO 🗋 Discarding.
+        //
+        await t
+            .click(MultiA)
+            .click(MultiB, withCmdClick)
+            .dragToElement(MultiA, Discarding);
+
+        //
+        // Go to 🗋 Tree multiselect, so we can check for stale metadata
+        // coming from the guest frame.
+        // We also need to reload to avoid Scenario #1.
+        //
+        await Page.goToPage('Tree multiselect');
+        await t.eval(() => location.reload(true));
+        await Page.waitForIframeLoading();
+
+        //
+        // Discard the move change and wait for the guest frame to reload plus
+        // some extra timeout, so there's a chance for stale metadata to
+        // overwrite the tree state.
+        //
+        await PublishDropDown.discardAll();
+        await Page.waitForIframeLoading();
+        await t.wait(500);
+
+        //
+        // Assert that 🗋 MultiA and 🗋 MultiB can still be found.
+        //
+        await t
+            .expect(MultiA.exists)
+            .ok('🗋 MultiA has disappeared');
+        await t
+            .expect(MultiB.exists)
+            .ok('🗋 MultiB has disappeared');
+    }
+)
+
+test(
+    'Scenario #3: Discarding a move change while being on a moved node does not'
+    + ' lead to an error in the guest frame',
+    async t => {
+        //
+        // Select 🗋 MultiA and 🗋 MultiB, then drag both documents INTO 🗋 MultiC.
+        //
+        await t
+            .click(MultiA)
+            .click(MultiB, withCmdClick)
+            .dragToElement(MultiA, MultiC);
+
+        //
+        // Go to 🗋 Home and reload the backend, so we avoid Scenario #1's
+        // underlying issue.
+        //
+        await Page.goToPage('Home');
+        await t.eval(() => location.reload(true));
+
+        //
+        // Go to 🗋 MultiA, so we are on a moved node in the guest frame.
+        //
+        await Page.goToPage('MultiA');
+
+        //
+        // Discard the move change.
+        //
+        await PublishDropDown.discardAll();
+
+        //
+        // Assert that there's no error showing up in the guest frame and
+        // that we're instead seeing the next-higher document node.
+        //
+        await Page.waitForIframeLoading();
+        await t.switchToIframe('[name="neos-content-main"]');
+        await t
+            .expect(Selector('.neos-error-screen').exists)
+            .notOk('There\'s an unexpected error screen in the guest frame.');
+
+        await t.switchToMainWindow();
+        await t
+            .expect(Selector('[role="treeitem"] [role="button"][class*="isFocused"]').textContent)
+            .eql('MultiC');
```

Comment:
thank you so much for adding the tests. I just ran them on 9.0 to check if we need any change there.
The first two scenarios pass but the third has at the end a wrong assertion, or 9.0 has a bug:

Instead of C being focused `MultiA` is:

<img width="453" alt="image" src="https://github.com/neos/neos-ui/assets/85400359/4d6e139a-e690-4671-abed-811c890d4e0a">

Reply:
And this seems to be okay from looking at it, as having C highlighted makes no sense as well and in your video its a bit glitchy :D

<img width="308" alt="image" src="https://github.com/neos/neos-ui/assets/85400359/56d8ea17-96b6-4b4c-aca1-e1deb0ff6390">

So i think we can just change the assertion when up-merging.